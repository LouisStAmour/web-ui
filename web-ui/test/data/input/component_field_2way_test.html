<!doctype html>
<!--
   Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
   for details. All rights reserved. Use of this source code is governed by a
   BSD-style license that can be found in the LICENSE file.
-->
<html lang="en">
<head>
  <meta charset="utf-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
  <script src="packages/web_ui/testing/testing.js"></script>
  <style>x-greeter { display: block; }</style>
</head>
<body>
  <element name="x-greeter" extends="div" constructor="Greeter">
    <template></template>
    <script type='application/dart'>
      import 'package:web_ui/web_ui.dart';
      @observable
      class Greeter extends WebComponent {
        String greetIng;

        // TODO(jmesserly): original test was testing for the "title" DOM
        // property. Unfortunately it can't be observed, so we'll use
        // a different property.
        String title2;

        changeStuff() {
          greetIng = 'Yo dawg';
          title2 = 'I heard you like $title2 so we put $title2 in yo $title2.';
        }
      }
    </script>
  </element>
  <p>
  The message changes from "hello world" to something more interesting.
  </p>
  <!-- Note: "title" is a DOM field on all elements.
       Also this element has no rendering. -->
  <x-greeter id="greet" bind-title2="data[1]" bind-greet-ing="data[0]">
  </x-greeter>
  <p id="actual">{{data[0]}} {{data[1]}}</p>
  <script type="application/dart">
    import 'dart:async';
    import 'dart:html';
    import 'package:unittest/unittest.dart';
    import 'package:web_ui/web_ui.dart';

    final data = toObservable(['hello', 'world']);

    main() {
      Timer.run(() {
        var actual = query('#actual');
        var greet = query('#greet');
        expect(actual.innerHtml, 'hello world');
        greet.xtag.changeStuff();
        deliverChangesSync();
        expect(actual.innerHtml, contains('Yo dawg I heard'));
        window.postMessage('done', '*');
      });
    }
  </script>
</body>
</html>
