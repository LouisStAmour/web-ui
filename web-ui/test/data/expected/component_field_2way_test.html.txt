Content-Type: text/plain
layer at (0,0) size 800x600
  RenderView at (0,0) size 800x600
layer at (0,0) size 800x84
  RenderBlock {HTML} at (0,0) size 800x84
    RenderBody {BODY} at (8,16) size 784x52
      RenderBlock {P} at (0,0) size 784x18
        RenderText {#text} at (0,0) size 690x18
          text run at (0,0) width 690: "The message changes from \"hello world\" to something more interesting."
      RenderBlock {DIV} at (0,34) size 784x0
      RenderBlock {P} at (0,34) size 784x18
        RenderText {#text} at (0,0) size 70x18
          text run at (0,0) width 70: "Yo dawg"
        RenderText {#text} at (70,0) size 10x18
          text run at (70,0) width 10: " "
        RenderText {#text} at (80,0) size 510x18
          text run at (80,0) width 510: "I heard you like world so we put world in yo world."
